src/core/services/google-maps-types.ts

```diff
@@ -191,3 +191,65 @@ export interface InfoWindowOptions {
   position?: LatLng|LatLngLiteral;
   zIndex?: number;
 }
+
+export interface Point {
+  x: number;
+  y: number;
+  equals(other: Point): boolean;
+  toString(): string;
+}
+
+export interface GoogleSymbol {
+  anchor?: Point;
+  fillColor?: string;
+  fillOpacity?: string;
+  labelOrigin?: Point;
+  path?: string;
+  rotation?: number;
+  scale?: number;
+  strokeColor?: string;
+  strokeOpacity?: number;
+  strokeWeight?: number;
+}
+
+export interface IconSequence {
+  fixedRotation?: boolean;
+  icon?: GoogleSymbol;
+  offset?: string;
+  repeat?: string;
+}
+
+export interface PolylineOptions {
+  clickable?: boolean;
+  draggable?: boolean;
+  editable?: boolean;
+  geodesic?: boolean;
+  icon?: Array<IconSequence>;
+  map?: GoogleMap;
+  path?: Array<LatLng>|Array<LatLng|LatLngLiteral>;
+  strokeColor?: string;
+  strokeOpacity?: number;
+  strokeWeight?: number;
+  visible?: boolean;
+  zIndex?: number;
+}
+
+export interface Polyline extends MVCObject {
+  getDraggable(): boolean;
+  getEditable(): boolean;
+  getMap(): GoogleMap;
+  getPath(): Array<LatLng>;
+  getVisible(): boolean;
+  setDraggable(draggable: boolean): void;
+  setEditable(editable: boolean): void;
+  setMap(map: GoogleMap): void;
+  setOptions(options: any): void;
+  setPath(path: Array<LatLng|LatLngLiteral>): void;
+  setVisible(visible: boolean): void;
+}
+
+export interface PolyMouseEvent extends MouseEvent {
```

Comment:
I've added the comment to google-map-types and exported it from there. I hope that works for the documentation since I'd rather avoid duplicating types.